crates/networking/discv5/src/subnet.rs

```diff
@@ -0,0 +1,141 @@
+use alloy_rlp::bytes::Bytes;
+use discv5::Enr;
+use ssz::{Decode, Encode};
+use ssz_types::{BitVector, typenum::U64};
+use tracing::trace;
+
+pub const ATTESTATION_BITFIELD_ENR_KEY: &str = "attnets";
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Subnet {
+    Attestation(u8),
+    SyncCommittee(u8),
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct Subnets {
+    attestation_bits: Option<BitVector<U64>>,
+}
+
+impl Subnets {
+    pub fn new() -> Self {
+        Self {
+            attestation_bits: None,
+        }
+    }
+
+    pub fn enable_subnet(&mut self, subnet: Subnet) -> Result<(), String> {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => {
+                let bits = self.attestation_bits.get_or_insert(BitVector::new());
+                bits.set(id as usize, true)
+                    .map_err(|_| "Subnet ID out of bounds".to_string())?;
+                Ok(())
+            }
+            Subnet::Attestation(_) => Ok(()),
+            Subnet::SyncCommittee(_) => unimplemented!("SyncCommittee support not yet implemented"),
+        }
+    }
+
+    pub fn disable_subnet(&mut self, subnet: Subnet) -> Result<(), String> {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => {
+                if let Some(bits) = &mut self.attestation_bits {
+                    bits.set(id as usize, false)
+                        .map_err(|_| "Subnet ID out of bounds".to_string())?;
+                }
+                Ok(())
+            }
+            Subnet::Attestation(_) => Ok(()),
+            Subnet::SyncCommittee(_) => unimplemented!("SyncCommittee support not yet implemented"),
+        }
+    }
+
+    pub fn is_active(&self, subnet: Subnet) -> bool {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => self
+                .attestation_bits
+                .as_ref()
+                .is_some_and(|bits| bits.get(id as usize).unwrap_or(false)),
+            Subnet::Attestation(_) => false,
+            Subnet::SyncCommittee(_) => unimplemented!("SyncCommittee support not yet implemented"),
+        }
+    }
+
+    pub fn attestation_bytes(&self) -> Option<Bytes> {
+        self.attestation_bits
+            .as_ref()
+            .map(|bits| bits.as_ssz_bytes().into())
+    }
+
+    pub fn from_enr(enr: &Enr) -> Result<Subnets, String> {
+        let bitfield_bytes: Bytes = enr
+            .get_decodable(ATTESTATION_BITFIELD_ENR_KEY)
+            .ok_or("ENR attestation bitfield non-existent")?
+            .map_err(|_| "Invalid RLP Encoding")?;
+
+        let attestation_bits = BitVector::from_ssz_bytes(&bitfield_bytes)
+            .map_err(|_| "Could not decode the ENR attnets bitfield")?;
+        Ok(Subnets {
+            attestation_bits: Some(attestation_bits),
+        })
+    }
+}
+
+pub fn subnet_predicate(subnets: Vec<Subnet>) -> impl Fn(&Enr) -> bool + Send + Sync {
```

Comment:
```suggestion
pub fn subnet_predicate(subnets: &[Subnet]) -> impl Fn(&Enr) -> bool + Send + Sync {
```

Reply:
It's bit tricky to pass reference here because `subnet_predicate` returns a closure which is boxed later and passed to `find_node_predicate`  function so for reference it's hard to address 'static lifetime requirement which is imposed by `find_node_predicate`